torch_geometric/nn/conv/rgcn_conv.py

```diff
@@ -209,14 +215,23 @@ def forward(self, x: Union[OptTensor, Tuple[OptTensor, Tensor]],
                 out += h.contiguous().view(-1, self.out_channels)
 
         else:  # No regularization/Basis-decomposition ========================
-            for i in range(self.num_relations):
-                tmp = masked_edge_index(edge_index, edge_type == i)
-
-                if x_l.dtype == torch.long:
-                    out += self.propagate(tmp, x=weight[i, x_l], size=size)
-                else:
-                    h = self.propagate(tmp, x=x_l, size=size)
-                    out = out + (h @ weight[i])
+            if not pyg_lib_avail:
+                for i in range(self.num_relations):
+                    tmp = masked_edge_index(edge_index, edge_type == i)
+
+                    if x_l.dtype == torch.long:
+                        out += self.propagate(tmp, x=weight[i, x_l], size=size)
+                    else:
+                        h = self.propagate(tmp, x=x_l, size=size)
+                        out = out + (h @ weight[i])
+            else:
+                edge_type, sort_by_edge_type = torch.sort(edge_type)
```

Comment:
We should only sorry if necessary, e.g., either via a is_sorted flag or via `edge_type[1:] >= edge_type[:-1].